include/toolchain.h

```diff
@@ -35,13 +35,16 @@
 
 #if defined(__XCC__)
 #include <toolchain/xcc.h>
+#define ZRESTRICT restrict
 #elif defined(__GNUC__) || (defined(_LINKER) && defined(__GCC_LINKER_CMD__))
 #include <toolchain/gcc.h>
+#define ZRESTRICT __restrict
```

Comment:
this should be defined in the respective toolchain header file, not in this top-level header.

Reply:
xcc.h actually includes gcc.h which includes common.h.  Also, other.h not provided, so I need some more clarification on the best way to do this.

Is it ok for #define ZRESTRICT to stay along side other.h in this top-level header?
Do I just #define ZRESTRICT __restrict in gcc.h and let that take care of gcc and xcc?
Do I then #ifndef #define ZRESTRICT restrict #endif in common.h?  Will that be enough?

Reply:
@nashif whats still needed to get this merged? It would be great to have this fixed